### PR TITLE
Document significance and creation of trusted clients

### PR DESCRIPTION
**Pull request by rlucioni**
FYI @edx/ecommerce

### CODE REVIEW

**Comment by clintonb** [comment]
:+1:

File: {{cookiecutter.repo_name}}/docs/getting_started.rst [hunk 1]
```diff
@@ -46,6 +46,11 @@ A new OAuth 2.0 client can be created at ``http://127.0.0.1:8000/admin/oauth2/cl
     7. Select :guilabel:`Confidential (Web applications)` as the client type.
     8. Click :guilabel:`Save`.
 
+Your newly-created client must also be designated as trusted. Trusted clients bypass the user consent form typically displayed after validating the user's credentials. Create a new trusted client at ``http://127.0.0.1:8000/admin/oauth2_provider/trustedclient/add/``.
+
```
Comment (nedbat) [inline {{cookiecutter.repo_name}}/docs/getting_started.rst]: Can we keep the lines hard-wrapped? (Ideally narrower than the 120 characters in the rest of this file...)

Comment (rlucioni) [inline {{cookiecutter.repo_name}}/docs/getting_started.rst]: @nedbat updated.

File: {{cookiecutter.repo_name}}/docs/getting_started.rst [hunk 2]
```diff
@@ -46,6 +46,12 @@ A new OAuth 2.0 client can be created at ``http://127.0.0.1:8000/admin/oauth2/cl
     7. Select :guilabel:`Confidential (Web applications)` as the client type.
     8. Click :guilabel:`Save`.
 
+Your newly-created client must also be designated as trusted. Trusted clients bypass the user consent form typically displayed after
```
Comment (nedbat) [inline {{cookiecutter.repo_name}}/docs/getting_started.rst]: 133 characters? :(

**Comment by rlucioni** [comment]
@nedbat I think Pylint 1.5.2, and by extension edx-lint 0.4.1, may be broken. Installing the latest version of edx-lint causes this PR's build to fail with an AttributeError. Could https://github.com/edx/edx-lint/pull/31 be at fault? Pinning edx-lint to 0.4.0 here causes some new quality violations to be identified, but allows the quality checks to run successfully.

File: {{cookiecutter.repo_name}}/docs/getting_started.rst (hunk 2 above)
Comment (rlucioni) [inline {{cookiecutter.repo_name}}/docs/getting_started.rst]: Updated.

**Comment by rlucioni** [comment]
@nedbat or @clintonb please review. edx-lint 0.4.1 is broken, so I've pinned to the last known working version, 0.4.0, and made changes necessary to appease Pylint.

**Comment by nedbat** [comment]
@rlucioni can you show how you referenced edx-lint before, and what attribute error you got?

**Comment by rlucioni** [comment]
@nedbat here's [an example](https://travis-ci.org/edx/cookiecutter-django-ida/builds/98171115) of the attribute error.

```
pylint --rcfile=pylintrc repo_name *.py
Traceback (most recent call last):
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/bin/pylint", line 11, in <module>
    sys.exit(run_pylint())
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/pylint/__init__.py", line 23, in run_pylint
    Run(sys.argv[1:])
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/pylint/lint.py", line 1332, in __init__
    linter.check(args)
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/pylint/lint.py", line 747, in check
    self._do_check(files_or_modules)
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/pylint/lint.py", line 869, in _do_check
    self.check_astroid_module(ast_node, walker, rawcheckers, tokencheckers)
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/pylint/lint.py", line 946, in check_astroid_module
    walker.walk(ast_node)
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/pylint/utils.py", line 874, in walk
    self.walk(child)
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/pylint/utils.py", line 871, in walk
    cb(astroid)
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/pylint/checkers/typecheck.py", line 293, in visit_assign
    function_node = safe_infer(node.value.func)
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/pylint/checkers/utils.py", line 92, in safe_infer
    value = next(inferit)
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/astroid/bases.py", line 327, in wrapped
    for res in _func(node, context, **kwargs):
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/astroid/bases.py", line 351, in wrapper
    for node in func(*args, **kwargs):
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/astroid/inference.py", line 241, in infer_getattr
    for owner in self.expr.infer(context):
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/astroid/bases.py", line 86, in cache_generator
    for result in generator:
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/astroid/bases.py", line 327, in wrapped
    for res in _func(node, context, **kwargs):
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/astroid/bases.py", line 124, in _infer_stmts
    for infered in stmt.infer(context):
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/astroid/bases.py", line 86, in cache_generator
    for result in generator:
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/astroid/bases.py", line 327, in wrapped
    for res in _func(node, context, **kwargs):
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/astroid/bases.py", line 124, in _infer_stmts
    for infered in stmt.infer(context):
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/astroid/bases.py", line 86, in cache_generator
    for result in generator:
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/astroid/bases.py", line 327, in wrapped
    for res in _func(node, context, **kwargs):
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/astroid/inference.py", line 229, in infer_from
    module = self.do_import_module()
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/astroid/mixins.py", line 107, in do_import_module
    return mymodule.import_module(modname, level=level)
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/astroid/scoped_nodes.py", line 431, in import_module
    return MANAGER.ast_from_module_name(modname)
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/astroid/manager.py", line 160, in ast_from_module_name
    return self.ast_from_file(filepath, modname, fallback=False)
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/astroid/manager.py", line 112, in ast_from_file
    return AstroidBuilder(self).file_build(filepath, modname)
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/astroid/builder.py", line 134, in file_build
    module = self._data_build(data, modname, path)
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/astroid/builder.py", line 177, in _data_build
    module = rebuilder.visit_module(node, modname, node_file, package)
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/astroid/rebuilder.py", line 150, in visit_module
    return self._transform(newnode)
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/astroid/manager.py", line 337, in transform
    ret = transform_func(node)
  File "/home/travis/build/edx/cookiecutter-django-ida/repo_name/.venv/local/lib/python2.7/site-packages/pylint_django/transforms/__init__.py", line 29, in set_fake_locals
    module._locals[class_name] = fake._locals[class_name]  # pylint: disable=protected-access
AttributeError: 'Module' object has no attribute '_locals'
```

Looks like `pylint_django` might be at fault.

I'm not sure what you mean by "how you referenced edx-lint before." It was included in the project's requirements file without being pinned to a version.

**Comment by nedbat** [comment]
@rlucioni OK, that's because the old requirements capped the versions of pylint and astroid, but not edx-lint, so who knows what would happen?  You could try unpinning edx-lint, and removing all mention of pylint and astroid.

**Comment by rlucioni** [comment]
@nedbat good call, I've tried that and it appears to work. I had assumed that since edx-lint was listed after pylint and astroid in the old requirements, it was overriding the pinned versions of those requirements during installation. I've observed such behavior before when working with other packages.

Regardless, everything seems to be in order now.

**Pull request merged**